crates/wasmtime/src/runtime/vm/instance/allocator/pooling.rs

```diff
@@ -644,14 +645,29 @@ unsafe impl InstanceAllocator for PoolingInstanceAllocator {
         self.live_core_instances.fetch_sub(1, Ordering::AcqRel);
     }
 
-    fn allocate_memory(
+    async fn allocate_memory(
         &self,
-        request: &mut InstanceAllocationRequest,
+        request: &mut InstanceAllocationRequest<'_>,
         ty: &wasmtime_environ::Memory,
-        tunables: &Tunables,
         memory_index: Option<DefinedMemoryIndex>,
     ) -> Result<(MemoryAllocationIndex, Memory)> {
-        self.with_flush_and_retry(|| self.memories.allocate(request, ty, tunables, memory_index))
+        // FIXME(rust-lang/rust#145127) this should ideally use a version of
+        // `with_flush_and_retry` but adapted for async closures instead of only
+        // sync closures. Right now that won't compile though so this is the
+        // manually expanded version of the method.
```

Comment:
Add a note for this to our items to discuss with the lang team, if we don't have one already?

Reply:
Oh this one's technically [already fixed](https://github.com/rust-lang/rust/pull/143545) and just waiting on stabilization, and browsing https://github.com/rust-lang/rust/issues/110338 shows this is pretty well-known, so not a major issue.